.test-infra/jenkins/job_beam_PreCommit_Java_GradleBuild.groovy

```diff
@@ -0,0 +1,62 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import common_job_properties
+
+// This is the Java precommit which runs a Gradle build, and the current set
+// of precommit tests.
+job('beam_PreCommit_Java_GradleBuild') {
+  description('Runs a build of the current GitHub Pull Request.')
+
+  // Execute concurrent builds if necessary.
+  concurrentBuild()
+
+  // Set common parameters.
+  common_job_properties.setTopLevelMainJobProperties(
+    delegate,
+    'master',
+    240)
+
+  def root_projects = [
+    ':beam-sdks-parent:beam-sdks-java-parent:beam-sdks-java-core',
+    ':beam-runners-parent:beam-runners-direct-java',
+    ':beam-sdks-parent:beam-sdks-java-parent:beam-sdks-java-fn-execution',
+  ]
+  def gradle_switches = [
+    // Continue the build even if there is a failure to show as many potential failures as possible.
+    '--continue',
+    // Until we verify the build cache is working appropriately, force rerunning all tasks
+    '--rerun-tasks',
+  ]
+
+  def gradle_command_line = './gradlew ' + gradle_switches.join(' ') + ' ' + root_projects.join(':buildNeeded ') + ' ' + root_projects.join(':buildDependents ')
```

Comment:
Prior to launching a command like this, you will need to change the current dir. Or you could/should build a command line that works from outside the checked out repo.